cognite_toolkit/cdf_tk/load.py

```diff
@@ -440,20 +471,21 @@ def load_resource(self, filepath: Path, dry_run: bool) -> Group:
                         values["scope"]["extractionPipelineScope"]["ids"] = [-1]
         return Group.load(raw)
 
-    def retrieve(self, ids: Sequence[int]) -> T_ResourceList:
-        remote = self.client.iam.groups.list(all=True).data
+    def retrieve(self, ids: SequenceNotStr[str]) -> GroupList:
+        remote = self.client.iam.groups.list(all=True)
         found = [g for g in remote if g.name in ids]
-        return found
+        return GroupList(found)
 
-    def delete(self, ids: Sequence[int], drop_data: bool) -> int:
+    def delete(self, ids: SequenceNotStr[str], drop_data: bool) -> int:
+        ids_ = list(ids)
```

Comment:
What is the idea behind an _ suffix?

Reply:
`mypy` does not allow you to change the type of a variable, so since you need `ids` later to be a mutable sequence, I usea a new variable for it. 

Renamed it to `id_list` to make it hopefully easier.